packages/hardhat-core/test/builtin-tasks/flatten.ts

```diff
@@ -412,6 +414,21 @@ describe("Flatten task", () => {
       assert(!spyFunctionConsoleWarn.called);
     });
 
+    it("should write to an output file when the parameter output is specified", async function () {
+      const outputFile = `${tmpdir()}/flatten.sol`;
+      try {
+        await this.env.run(TASK_FLATTEN, {
+          files: ["contracts/A.sol", "contracts/D.sol"],
+          output: outputFile,
+        });
+        const expected = await getExpectedSol();
+        const actual = readFileSync(outputFile, "utf8");
+        assert.equal(actual, expected);
+      } finally {
+        removeSync(outputFile);
```

Comment:
I put this in a try/finally block to ensure we clean up the temp file on failed runs, too.